ariadne_codegen/contrib/shorter_results.py

```diff
@@ -0,0 +1,369 @@
+
+    class GetAuthenticatedUser(BaseModel):
+        me: Optional["GetAuthenticatedUserMe"]
+
+When later using the client to call `get_authenticated_user` the return type
+would most of the time immediately be expanded to get the inner
+`GetAuthenticatedUserMe`:
+
+    me = (await client.get_authenticated_user).me
+
+By enabling this plugin the return type of the generated
+`get_authenticated_user` would instead directly return a
+`GetAuthenticatedUserMe` so the caller could use:
+
+    me = await client.get_authenticated_user
+
+This plugin can be enabled by either adding the plugin in the settings:
+
+    plugins = ["ariadne_codegen.ShorterResultsPlugin"]
+"""
+
+import ast
+from copy import deepcopy
+from typing import Union, Dict, Optional, List
+from graphql import (
+    GraphQLSchema,
+    ExecutableDefinitionNode,
+    SelectionSetNode,
+)
+from ariadne_codegen.codegen import (
+    generate_async_for,
+    generate_attribute,
+    generate_expr,
+    generate_name,
+    generate_return,
+    generate_subscript,
+    generate_yield,
+)
+
+from ariadne_codegen.plugins.base import Plugin
+
+
+class ShorterResultsPlugin(Plugin):
+    """
+    Make single field return types expanded.
+
+    All client method that returns type with a single field will be expanded and
+    instead return the type of the inner field.
+    """
+
+    def __init__(self, schema: GraphQLSchema, config_dict: Dict) -> None:
+        self.class_dict: Dict[str, ast.ClassDef] = {}
+        self.extended_imports: Dict[str, set] = {}
+
+        super().__init__(schema, config_dict)
+
+    def generate_result_class(
+        self,
+        class_def: ast.ClassDef,
+        operation_definition: ExecutableDefinitionNode,
+        selection_set: SelectionSetNode,
+    ) -> ast.ClassDef:
+        """Store a map of all classes and its ast"""
+        self.class_dict[class_def.name] = class_def
+
+        return super().generate_result_class(
+            class_def, operation_definition, selection_set
+        )
+
+    def generate_client_module(self, module: ast.Module) -> ast.Module:
+        """
+        Update the generated client.
+
+        At this point we've the full ast for all classes used for the client so
+        instead of modifying class by class in `generate_client_method` we
+        modify each method from here. By doing so we ensure that we have a map
+        between all classes, including fragments, so we can traverse them to
+        figure out how many fields we have in total for each request.
+
+        If we only have a single field, including when expanding our fragments,
+        we can expand the single field and update the return type.
+        """
+        client_def = next(
+            filter(lambda o: isinstance(o, ast.ClassDef), module.body), None
+        )
+        if not client_def:
+            return super().generate_client_module(module)
+
+        for method_def in [
+            m
+            for m in client_def.body
```

Comment:
Mypy fails on this line:
`ariadne_codegen/contrib/shorter_results.py:106: error: "stmt" has no attribute "body"  [attr-defined]`

Reply:
Fixed by asserting that `client_def` is `ast.ClassDef`